IAmLegend/Assets/Scripts/Gun.cs

```diff
@@ -31,12 +33,22 @@ public void Shoot()
                 {
                     Instantiate(bulletPrefab, position, rotation * Quaternion.Euler(Vector3.up * currentAngle));
                     currentAngle += distanceBetweenBullets;
+                    muzzleFlash.SetActive(true);
```

Comment:
you add the muzzle flash 3 times (because of for loop) for the shotgun but it doesn't really make a difference (as far as I can tell).

So it might be simpler to just remove the duplicated calls to 
muzzleFlash.SetActive(true);
StartCoroutine(MuzzleFlashWait());

and just move it after the else block